### PR TITLE
comment model added

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
Blog App Backend:

Description: Comment model updated. It now allows multithreaded commenting like Facebook. it's relationship with Blog and User and it's serializer has been added.

Installation Instructions:
1 .Change directory to `webblog/` and make a virtual environment by running `virtualenv env`
2 .Open terminal and start virtual environment by running `source ./env/bin/activate`
3 .run `pip install -r requirements.txt --no-index --find-links file:///tmp/packages` to make requirements
4. run `python manage.py runserver` to start the django server

### CODE REVIEW

**Comment by asadmanzoor93** [comment]
@Ali-D-Akbar replace PR title with proper descriptive one.

File: Backend/webblog/accounts/views.py [hunk 1]
```diff
@@ -15,8 +15,8 @@ def post(self, request, *args, **kwargs):
         serializer.is_valid(raise_exception=True)
         user = serializer.save()
         return Response({
-                "user": UserSerializer(user, context=self.get_serializer_context()).data,
-                "token": AuthToken.objects.create(user)[1]
+            "user": UserSerializer(user, context=self.get_serializer_context()).data,
+            "token": AuthToken.objects.create(user, )[1]
```
Comment (asadmanzoor93) [inline Backend/webblog/accounts/views.py]: Remove this comma.

File: Backend/webblog/accounts/views.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def post(self, request, *args, **kwargs):
         user = serializer.validated_data
         return Response({
             "user": UserSerializer(user, context=self.get_serializer_context()).data,
-            "token": AuthToken.objects.create(user)[1]
+            "token": AuthToken.objects.create(user, )[1]
```
Comment (asadmanzoor93) [inline Backend/webblog/accounts/views.py]: Why this comma added while creating AuthToken across multiple files ?

File: Backend/webblog/blog/models.py [hunk 3]
```diff
@@ -6,3 +6,10 @@ class Blog(models.Model):
     description = models.TextField(max_length=1500)
```
Comment (zubair-arbi) [inline Backend/webblog/blog/models.py]: every blog post should have and image

File: Backend/webblog/blog/models.py [hunk 4]
```diff
@@ -6,3 +6,10 @@ class Blog(models.Model):
...
     created = models.DateTimeField(auto_now_add=True)
     owner = models.ForeignKey('auth.User', related_name='blog', on_delete=models.CASCADE, null=True)
+
+
+class Comment(models.Model):
+    blog = models.ForeignKey(Blog, related_name='comment', on_delete=models.CASCADE)
+    description = models.TextField(max_length=1500)
+    created = models.DateTimeField(auto_now_add=True)
```
Comment (zubair-arbi) [inline Backend/webblog/blog/models.py]: use  `TimeStampedModel` from `django-extensions` (https://github.com/django-extensions/django-extensions)
`from django_extensions.db.models import TimeStampedModel`

File: Backend/webblog/blog/models.py [hunk 5]
```diff
@@ -6,3 +6,10 @@ class Blog(models.Model):
...
+    owner = models.ForeignKey('auth.User', related_name='comment', on_delete=models.CASCADE, null=True)
```
Comment (zubair-arbi) [inline Backend/webblog/blog/models.py]: we should consider how we will handle threaded response.
Screen Shot 2019-09-24 at 11 12 40 AM

File: Backend/webblog/accounts/views.py (hunk 1 above)
Comment (Ali-D-Akbar) [reply Backend/webblog/accounts/views.py]: might've been added by the Pycharm.
it's been removed now.

File: Backend/webblog/blog/models.py (hunk 3 above)
Comment (Ali-D-Akbar) [reply Backend/webblog/blog/models.py]: Right sir, I'd be working on it! :+1:

File: Backend/webblog/blog/models.py (hunk 4 above)
Comment (Ali-D-Akbar) [reply Backend/webblog/blog/models.py]: This is causing many problems and I can't implement it properly.

File: Backend/webblog/blog/models.py (hunk 3 above)
Comment (Ali-D-Akbar) [reply Backend/webblog/blog/models.py]: image object has been added

File: Backend/webblog/accounts/views.py (hunk 2 above)
Comment (Ali-D-Akbar) [reply Backend/webblog/accounts/views.py]: might've been added by the Pycharm.
it's been removed now.

File: Backend/webblog/blog/models.py (hunk 5 above)
Comment (Ali-D-Akbar) [reply Backend/webblog/blog/models.py]: it has been implemented now.

File: Backend/webblog/blog/views.py [hunk 6]
```diff
@@ -1,29 +1,50 @@
...
-
     def perform_create(self, serializer):
         serializer.save(owner=self.request.user)
 
+    def perform_update(self, serializer):
+        serializer.save(created=timezone.now())
+
+    def list(self, request, *args, **kwargs):
+        if request.GET.get('title', '') != '':
+            queryset = Blog.objects.filter(title__icontains=request.GET.get('title'))
```
Comment (zubair-arbi) [inline Backend/webblog/blog/views.py]: From fronend we should expect only one argument `search_keyword` and on backend here we should find all the blogs which have those keywords in title or in the description.

Hint: Use Q objects for query: https://docs.djangoproject.com/en/2.2/topics/db/queries/#complex-lookups-with-q-objects

Comment (Ali-D-Akbar) [reply Backend/webblog/blog/views.py]: Done.
I've committed the changes.

**Comment by zubair-arbi** [comment]
@Ali-D-Akbar 

**we should only merge the commits with proper messages.** We should always squash the commits and merge the verbose commits only and avoid merging commits with useless or duplicate messages. For example we have merged following commits:

1. comment model added
2. multithreaded comments added	
3. multithreaded comments added
4. search improvements
5. search improvements

Instead of all these 5 commits we could have squashed them into one commit:
1. add multithreaded comments for blog posts and optimize search

**Also we should always delete the branches after merge.**

**Pull request merged**